src/core/config/local_config.ts

```diff
@@ -0,0 +1,188 @@
+
+            for(const key in parsedConfig) {
+                if (!allowedKeys.includes(key)) {
+                    throw new SoloError('Validation of local config failed')
+                }
+                this[key] = parsedConfig[key]
+            }
+
+            this.validate()
+            this.skipPromptTask = true
+        }
+    }
+
+    private validate () {
+        const genericMessage = 'Validation of local config failed'
+        const errors = validateSync(this, {})
+
+        if (errors.length) {
+            throw new SoloError(genericMessage)
+        }
+
+        try {
+            // Custom validations:
+            for (const deploymentName in this.deployments) {
+                const deployment = this.deployments[deploymentName]
+                const deploymentIsObject = deployment && typeof deployment === 'object'
+                const deploymentHasClusterAliases = deployment.clusterAliases && Array.isArray(deployment.clusterAliases)
+                let clusterAliasesAreStrings = true
+                for (const clusterAlias of deployment.clusterAliases) {
+                    if (typeof clusterAlias !== 'string') {
+                        clusterAliasesAreStrings = false
+                    }
+                }
+
+                if (!deploymentIsObject || !deploymentHasClusterAliases || !clusterAliasesAreStrings) {
+                    throw new SoloError(genericMessage)
+                }
+            }
+
+            for (const clusterName in this.clusterMappings) {
+                const contextName = this.clusterMappings[clusterName]
+                if (typeof clusterName !== 'string' || typeof contextName !== 'string') {
+                    throw new SoloError(genericMessage)
+                }
+            }
+
+            if (!this.deployments[this.currentDeploymentName]) {
+                throw new SoloError(genericMessage)
+            }
+        }
+        catch(e: any) { throw new SoloError(genericMessage) }
+    }
+
+    public setUserEmailAddress (emailAddress: string): this {
+        this.userEmailAddress = emailAddress
+        this.validate()
+        return this
+    }
+
+    public setDeployments (deployments: Deployments): this {
+        this.deployments = deployments
+        this.validate()
+        return this
+    }
+
+    public setClusterMappings (clusterMappings: ClusterMapping): this {
+        this.clusterMappings = clusterMappings
+        this.validate()
+        return this
+    }
+
+    public setCurrentDeployment (deploymentName: string): this {
+        this.currentDeploymentName = deploymentName
+        this.validate()
+        return this
+    }
+
+    public getCurrentDeployment (): Deployment {
+        return this.deployments[this.currentDeploymentName]
+    }
+
+    private configFileExists (): boolean {
+        return fs.existsSync(this.filePath)
+    }
+
+    public async write (): Promise<void> {
+        const yamlContent = yaml.stringify({
```

Comment:
I would think we could just do a JSON.stringify and yaml.dump, was this a better approach?  might be the same number of lines, though.  The only thing is, that if we modify the class, then this has to be kept in sync.  but, if you did JSON.stringify, then yaml.dump, then any new attributes would be automatic.